sklearn/svm/_classes.py

```diff
@@ -1316,3 +1319,227 @@ def probA_(self):
+
+    fit_status_ : int
+        0 if correctly fitted, 1 otherwise (will raise warning)
+
+    Examples
+    --------
+    >>> from sklearn.svm import SVDD
+    >>> X = [[0], [0.44], [0.45], [0.46], [1]]
+    >>> clf = SVDD(gamma='auto').fit(X)
+    >>> clf.predict(X)
+    array([-1,  1,  1,  1, -1])
+    >>> clf.score_samples(X)  # doctest: +ELLIPSIS
+    array([0.5298..., 0.8047..., 0.8056..., 0.8061..., 0.4832...])
+
+    References
+    ----------
+    .. [1] Tax, D.M. and Duin, R.P., 2004. "Support vector data
+           description." Machine learning, 54(1), pp.45-66.
+           doi:10.1023/B:MACH.0000008084.60811.49
+
+    .. [2] Chang, W.C., Lee, C.P. and Lin, C.J., 2013. "A revisit
+           to support vector data description (SVDD)." Technical
+           Report, Department of Computer Science, National Taiwan
+           University.
+    """
+
+    _impl = 'svdd_l1'
+
+    def __init__(self, kernel='rbf', degree=3, gamma='scale',
+                 coef0=0.0, tol=1e-3, nu=0.5, shrinking=True, cache_size=200,
+                 verbose=False, max_iter=-1):
+
+        super().__init__(
+            kernel=kernel, degree=degree, gamma=gamma, coef0=coef0,
+            tol=tol, C=0., nu=nu, epsilon=0., shrinking=shrinking,
+            probability=False, cache_size=cache_size, class_weight=None,
+            verbose=verbose, max_iter=max_iter, random_state=None)
+
+    def fit(self, X, y=None, sample_weight=None, **params):
+        """Learns the soft minimum volume hypersphere around the sample X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Set of samples, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Per-sample weights. Rescale C per sample. Higher weights
+            force the classifier to put more emphasis on these points.
+
+        y : Ignored
+            not used, present for API consistency by convention.
+
+        Returns
+        -------
+        self : object
+
+        Notes
+        -----
+        If X is not a C-ordered contiguous array it is copied.
+
+        """
+        super().fit(X, np.ones(_num_samples(X)),
+                    sample_weight=sample_weight, **params)
+        self.offset_ = -self._intercept_
+        return self
+
+    def decision_function(self, X):
+        """Signed distance to the enveloping hypersphere.
+
+        Signed distance is positive for an inlier and negative for an outlier.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data matrix.
+
+        Returns
+        -------
+        dec : ndarray of shape (n_samples,)
+            Returns the decision function of the samples.
+        """
+        dec = self._decision_function(X).ravel()
+        return dec
```

Comment:
nitpick:

```python
return self._decision_function(X).ravel()
```

Reply:
Thank you! Resolved this and aligned with `OneClassSVM.score_samples()`. Methods look much more clear now.